code/modules/mob/living/carbon/human/species_types/vampire.dm

```diff
@@ -153,14 +153,19 @@
 		to_chat(caster, "<span class='warning'>You're already shapeshifted!</span>")
 		return
 
+	var/mob/living/carbon/human/human_caster = caster
+	if(!human_caster?.dna?.species?.id == SPECIES_VAMPIRE)
+		to_chat(caster, "<span class='warning'>You don't seem to be able to shapeshift..</span>")
+		return
```

Comment:
Actually, I think this check should be located in `/obj/effect/proc_holder/spell/targeted/shapeshift/bat/cast(list/targets,mob/user = usr)` before the parent call to stop other unwanted behaviours.

Also lines 151-154 are unreachable since `/obj/effect/proc_holder/spell/targeted/shapeshift/Restore` would be run instead, thus they should be removed.

Reply:
Mh, I don't think I want to stop them from transforming back (if they somehow get changed while in the mob holder, for example), just initiating the transform itself which Shapeshift() does, so having it in this proc should be fine imo.

151-154, while rare, can be reached if Shapeshift() gets called manually / by different sources, so better to have a failsafe regardless.

human_caster?.[etc] should be a sufficient check for human_caster not existing, e.g. them being non-humanoid / the typecast failing, no? Unless I misunderstood how the ? operator works there.

Reply:
> Mh, I don't think I want to stop them from transforming back (if they somehow get changed while in the mob holder, for example), just initiating the transform itself which Shapeshift() does, so having it in this proc should be fine imo.

the _`if(locate(/obj/shapeshift_holder) in targets[1])`_ line should take care of it.

> 151-154, while rare, can be reached if Shapeshift() gets called manually / by different sources, so better to have a failsafe regardless.
> 
> if(!human_caster?.[etc]) should be a sufficient check for human_caster not existing, e.g. them being non-humanoid / the typecast failing, no? Unless I misunderstood how the ? operator works there.

Reply:
updated it & moved the species check to cast(), that fine now?